pkg/menu/menu.go

```diff
@@ -199,6 +199,13 @@ func parsingMenuOption(labels []string, menu *widgets.List, input, warning *widg
 			// 1.input is a number;
 			// 2.the number does not exceed the index in the current page.
 			if err == nil && c >= first && c < last {
+				// if there is not specific warning for this entry, return it
+				// elsewise show the warning and continue
+				if customWarning != nil && customWarning[c] != "" {
```

Comment:
change customWarning != nil to 
len(customWarning) > c
otherwise you are blindly indexing into an array that may be too short.